pkg/peer/peer.go

```diff
@@ -98,17 +102,69 @@ func (p *Peer[ID]) SubscribeTo(track *webrtc.TrackLocalStaticRTP) error {
 	// Before these packets are returned they are processed by interceptors. For things
 	// like NACK this needs to be called.
 	go func() {
-		rtcpBuf := make([]byte, 1500)
 		for {
-			if _, _, rtcpErr := rtpSender.Read(rtcpBuf); rtcpErr != nil {
-				return
+			packets, _, err := rtpSender.ReadRTCP()
+			if err != nil {
+				if errors.Is(err, io.ErrClosedPipe) || errors.Is(err, io.EOF) {
+					return
+				}
+
+				p.logger.WithError(err).Warn("failed to read RTCP on track")
 			}
+
+			p.sink.Send(ForwardRTCP{Packets: packets, TrackID: track.ID(), StreamID: track.StreamID()})
 		}
 	}()
 
 	return nil
 }
 
+func (p *Peer[ID]) WriteRTCP(packets []rtcp.Packet, streamID string, trackID string, lastPLITimestamp int64) {
+	packetsToSend := []rtcp.Packet{}
+	var mediaSSRC uint32
+	for _, receiver := range p.peerConnection.GetReceivers() {
+		if receiver.Track().ID() == trackID && receiver.Track().StreamID() == streamID {
+			mediaSSRC = uint32(receiver.Track().SSRC())
+			break
+		}
+	}
```

Comment:
We probably want to return an error if we have not found a track with the given `trackID` and `streamID` (otherwise also the `WriteRTCP` below would fail (unknown track).

Reply:
Good point!